public/app/features/alerting/unified/Templates.tsx

```diff
@@ -1,31 +1,23 @@
-import { Route, Switch } from 'react-router-dom';
+import { Routes, Route } from 'react-router-dom-v5-compat';
 
 import { withErrorBoundary } from '@grafana/ui';
-import { SafeDynamicImport } from 'app/core/components/DynamicImports/SafeDynamicImport';
 
 import { AlertmanagerPageWrapper } from './components/AlertingPageWrapper';
-
-const EditMessageTemplate = SafeDynamicImport(() => import('./components/contact-points/EditMessageTemplate'));
-const NewMessageTemplate = SafeDynamicImport(() => import('./components/contact-points/NewMessageTemplate'));
-const DuplicateMessageTemplate = SafeDynamicImport(
-  () => import('./components/contact-points/DuplicateMessageTemplate')
-);
+import DuplicateMessageTemplate from './components/contact-points/DuplicateMessageTemplate';
+import EditMessageTemplate from './components/contact-points/EditMessageTemplate';
+import NewMessageTemplate from './components/contact-points/NewMessageTemplate';
 
 const NotificationTemplates = (): JSX.Element => (
   <AlertmanagerPageWrapper
     navId="receivers"
     accessType="notification"
     pageNav={{ id: 'templates', text: 'Notification templates', subTitle: 'Create and edit notification templates' }}
   >
-    <Switch>
-      <Route exact={true} path="/alerting/notifications/templates/:name/edit" component={EditMessageTemplate} />
-      <Route exact={true} path="/alerting/notifications/templates/new" component={NewMessageTemplate} />
-      <Route
-        exact={true}
-        path="/alerting/notifications/templates/:name/duplicate"
-        component={DuplicateMessageTemplate}
-      />
-    </Switch>
+    <Routes>
+      <Route path=":name/edit" element={<EditMessageTemplate />} />
+      <Route path="new" element={<NewMessageTemplate />} />
+      <Route path=":name/duplicate" element={<DuplicateMessageTemplate />} />
+    </Routes>
```

Comment:
Do we ideally want to hoist these to the main alerting routes file as well? I know you mentioned in the description that it's not required, but what is the end goal/pattern for managing routes? Or is it up to us (within the scope of Alerting)?

Reply:
These are proper child routes, as opposed to the duplicate routes in other places, so I think they can stay here. The main issue was with the routes defined separately and in the routes file, making their migration harder.